concurrency.go

```diff
@@ -99,7 +100,16 @@ func Async6[A, B, C, D, E, F any](f func() (A, B, C, D, E, F)) <-chan Tuple6[A,
 
 // WaitFor runs periodically until a condition is validated.
 func WaitFor(condition func(i int) bool, maxDuration time.Duration, tick time.Duration) (int, time.Duration, bool) {
```

Comment:
Also, I would like to use named variables there

These are too cryptic.


```suggestion
 func WaitFor(condition func(i int) bool, maxDelay time.Duration, interval time.Duration) (iteration int, elapsed time.Duration, timeoutReached bool) {
```

Reply:
What are your thoughts, any suggestions?

Reply:
sure!